### PR TITLE
Take player status into consideration for phalanxes in range

### PR DESCRIPTION
**Pull request by kejith**
image
image

Player with Phalanx is:
Inactive -> Grey
Umode || Banned -> Blue
Active -> Red

### CODE REVIEW

File: userscript/ui/tabStatus.js [hunk 1]
```diff
@@ -43,6 +48,40 @@ class GalaxyPhalanxInformation {
     return this.getSystem(system).length > 0
   }
 
+  getActivity (system) {
+    if (system < 1 || system > 400) return 3
+
+    const phalanxInSystem = this.getSystem(system)
+    let activity = 0
+    phalanxInSystem.forEach(phalanx => {
+      if (phalanx.galaxy === 4 && phalanx.system === 30) console.log({ phalanx, activity })
```
Comment (anoff) [inline userscript/ui/tabStatus.js:57]: whaaaaat?

File: userscript/ui/tabStatus.js [hunk 2]
```diff
@@ -43,6 +48,40 @@ class GalaxyPhalanxInformation {
     return this.getSystem(system).length > 0
   }
 
+  getActivity (system) {
+    if (system < 1 || system > 400) return 3
```
Comment (anoff) [inline userscript/ui/tabStatus.js:52]:
```suggestion
    if (system < 1 || system > 400) return ACTIVITY_ACTIVE
```

File: userscript/features/galaxyview.js [hunk 3]
```diff
@@ -370,8 +370,13 @@ function checkPhalanxesInRange (galaxy, system) {
...
         phalanxes.forEach(phalanx => {
           const span = document.createElement('span')
+
+          let color = 'red'
+          if (phalanx.isInactive > 0) color = '#999'
+          if (phalanx.isVacation || phalanx.isBanned) color = '#659ec7'
+
           span.innerHTML = `
-          <a style="color: red;"
+          <a style="color: ${color}; ${phalanx.isBanned ? 'text-decoration: line-through;' : ''}"
```
Comment (anoff) [inline userscript/features/galaxyview.js:379]: if you already decide to turn what can be done in 3 lines into a 23 line change, at least reuse the functions you created :P

**Review by anoff — CHANGES_REQUESTED** [review]

File: userscript/utils.js [hunk 4]
```diff
@@ -126,6 +126,103 @@ class Feature {
...
+      case Activity.INACTIVE:
+        return '#999'
+      case Activity.VACATION:
+      case Activity.BANNED:
+        return '#659ec7'
+    }
+  }
+
+  getSystemActivityColor (system) {
+    if (system < 1 || system > 400) return 'red'
```
Comment (anoff) [inline userscript/utils.js:207]: seems random to just return `'red'` here.
Why not let the downstream function handle the edge cases (which it already does).

File: userscript/utils.js [hunk 5]
```diff
@@ -126,6 +126,103 @@ class Feature {
...
+      } else {
+        for (let system = from; system <= (400 + to); system++) {
+          this.appendPhalanx(system % 401, phalanx)
+        }
+      }
+    })
+  }
+
+  isInRange (system) {
+    if (system < 1 || system > 400) return false
```
Comment (anoff) [inline userscript/utils.js:171]: please turn the `400` into a pseudo constant somewhere, too much magic numbers

File: userscript/utils.js [hunk 6]
```diff
@@ -126,6 +126,103 @@ class Feature {
...
+    if (!Array.isArray(phalanxes)) {
+      throw new Error('Expected an array of phalanx information.')
+    }
+    this.phalanxes = phalanxes
+    this.data = Array.from(Array(400), () => [])
+    this.calculateSystemsInPhalanx()
+  }
+
+  getSystem (system) {
+    return this.data[system - 1]
```
Comment (anoff) [inline userscript/utils.js:147]: If anywhere, doesn't this need the check for `1..400` bounds? :D

**Review by anoff — CHANGES_REQUESTED** [review]
Can you add some docstring pleaaaase?
It can be confusing to have methods with such similar names doing very different things `getActivity`, `getSystemActivity`, `getActivityColor`, `getSystemActivityColor`

File: userscript/utils.js (hunk 4 above)
Comment (kejith) [reply userscript/utils.js:207]: Something went really wrong when this case happens and just to be safe we dont send a false negative we assume the system is in "danger" and color red. 

In Production no Galaxy should ever see this happening.

**Review by anoff — APPROVED** [review]
LGTM thanks :)

**Pull request merged**